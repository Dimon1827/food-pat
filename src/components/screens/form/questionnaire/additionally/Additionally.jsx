import './Additionally.scss';
import AdditionallyItem from './AdditionallyItem';
import checkBoxItems from './checkBoxItem';

function Additionally({ control }) {

  return (
    <div className="container">
      <div className="questionnaire__field additionally">
        <h2 className="additionally__title form-title">Аллергены</h2>
        <ul className="additionally__wrapper">
          {checkBoxItems.length > 0 &&
            checkBoxItems.map((item) => (
              <AdditionallyItem
                key={item.id}
                control={control}
                type={item.type}
                {...item}
              />
            ))}
        </ul>
        <button type="submit" className="questionnaire__button button">
          отправить заявку
        </button>
        <p className="questionnaire__text">* — Обязательные поля</p>
      </div>
    </div>
  );
}

export default Additionally;
