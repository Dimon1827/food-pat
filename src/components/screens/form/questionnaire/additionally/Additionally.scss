@import 'src/assets/styles/variables';

.additionally {
  @media (min-width: $desktop-width) {
    padding-bottom: 20px;
  }
}

.additionally__wrapper {
  display: grid;
  padding: 0;
  margin: 0;
  list-style-type: none;
  margin-bottom: 26px;

  @media (min-width: $tablet-width) {
    grid-template-columns: repeat(2, 1fr);
    gap: 50px 60px;
    align-items: center;
    margin-bottom: 61px;
  }

  @media (min-width: $desktop-width) {
    grid-template-columns: repeat(4, 245px);
    gap: 40px 80px;
    align-items: start;
    margin-bottom: 67px;
  }
}

.additionally__item {
  position: relative;
  display: flex;
  justify-content: space-between;
  align-items: center;

  &:hover {
    .additionally__label::before {
      border-color: $special-grey-5;
    }
  }

  &:not(:last-child) {
    margin-bottom: 25px;
  }

  @media (min-width: $tablet-width) {
    &:not(:last-child) {
      margin-bottom: 0;
    }
  }
}

.additionally__title {
  margin-bottom: 40px;

  @media (min-width: $tablet-width) {
    margin-bottom: 66px;
  }
}

.additionally__label {
  padding-left: 45px;
  cursor: pointer;

  &::before {
    content: "";
    position: absolute;
    top: -5px;
    left: 0;
    width: 30px;
    height: 30px;
    background-color: $white-color;
    border: 1px solid $special-grey-2;

    @media (min-width: $tablet-width) {
      width: 40px;
      height: 40px;
      top: -9px;
    }
  }

  &::after {
    content: "";
    position: absolute;
    display: none;
    width: 10px;
    height: 16px;
    top: -2%;
    left: 10px;
    border: solid $bg-basic-orange;
    border-width: 0 2px 2px 0;
    transform: rotate(45deg);

    @media (min-width: $tablet-width) {
      width: 15px;
      height: 22px;
      top: -19%;
      left: 12px;
    }
  }

  &:not(:last-child) {
    margin-bottom: 24px;

    @media (min-width: $tablet-width) {
      margin-bottom: 0;
    }
  }

  @media (min-width: $tablet-width) {
    font-size: 20px;
    line-height: 20px;
    padding-left: 60px;
  }
}

.additionally__input:checked ~ .additionally__label::after {
  display: block;
}

.additionally__input:focus ~ .additionally__label::before {
  border-color: $special-grey-5;
}

.additionally__input:disabled ~ .additionally__label {
  cursor: default;
  opacity: 0.3;
}
