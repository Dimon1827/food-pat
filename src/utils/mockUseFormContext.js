import { useFormContext } from "react-hook-form";

export function mockUseFormContext(errors) {
  useFormContext.mockReturnValue({
    register: jest.fn(),
    formState: {
      errors: errors,
    },
  });
}


