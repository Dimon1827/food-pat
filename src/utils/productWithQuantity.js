const productWithQuantity = (food) => {
  return {
    ...food,
    isChecked: false,
    quantity: 1,
  };
};

export default productWithQuantity;
