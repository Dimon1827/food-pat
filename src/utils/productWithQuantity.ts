import { IProducts } from "redux/slice/productsSlice.types";

interface IproductWithQuantity extends Omit<IProducts, 'isChecked' | 'quantity'> {};

const productWithQuantity = (food: IproductWithQuantity) => {
  return {
    ...food,
    isChecked: false,
    quantity: 1,
  };
};

export default productWithQuantity;
