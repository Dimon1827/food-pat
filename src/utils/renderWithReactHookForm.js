import { render } from '@testing-library/react';
import { FormProvider, useForm } from 'react-hook-form';

export function renderWithReactHookForm(ui, { defaultValues = {} } = {}) {
  let reactHookFormMethods = {};

  const Wrapper = ({ children }) => {
    const methods = useForm({ defaultValues });
    return <FormProvider {...methods}>{children}</FormProvider>;
  };

  return {
    ...render(ui, { wrapper: Wrapper }),
  };
}