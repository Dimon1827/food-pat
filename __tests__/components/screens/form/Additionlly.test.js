import { screen } from '@testing-library/react';
import { renderWithReactHookForm } from '@/utils/renderWithReactHookForm';
import Additionally from '@/components/screens/form/questionnaire/additionally/Additionally';



describe('Additionally', () => {
  it('Additionally renders correctly as checkbox', () => {
    renderWithReactHookForm(<Additionally />);
    const list = screen.getByTestId('additionally');
    const listItem = screen.getAllByRole('listitem');
    const input = screen.getAllByRole('checkbox');
    expect(list).toMatchSnapshot();
    expect(listItem).toHaveLength(4);
    expect(input).toHaveLength(4);
  });
});
