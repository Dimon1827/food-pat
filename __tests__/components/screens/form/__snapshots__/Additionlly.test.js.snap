// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Additionally Additionally renders correctly as checkbox 1`] = `
<div
  class="questionnaire__field additionally"
  data-testid="additionally"
>
  <h2
    class="additionally__title form-title"
  >
    Аллергены
  </h2>
  <ul
    class="additionally__wrapper"
  >
    <li
      class="additionally__item"
    >
      <input
        class="visually-hidden control-input additionally__input"
        id="milk"
        name="milk"
        type="checkbox"
        value="milk"
      />
      <label
        class="label additionally__label"
        for="milk"
      >
        Молоко
      </label>
    </li>
    <li
      class="additionally__item"
    >
      <input
        class="visually-hidden control-input additionally__input"
        id="fish"
        name="fish"
        type="checkbox"
        value="fish"
      />
      <label
        class="label additionally__label"
        for="fish"
      >
        Рыба
      </label>
    </li>
    <li
      class="additionally__item"
    >
      <input
        class="visually-hidden control-input additionally__input"
        id="nuts"
        name="nuts"
        type="checkbox"
        value="nuts"
      />
      <label
        class="label additionally__label"
        for="nuts"
      >
        Орехи
      </label>
    </li>
    <li
      class="additionally__item"
    >
      <input
        class="visually-hidden control-input additionally__input"
        id="egg"
        name="egg"
        type="checkbox"
        value="egg"
      />
      <label
        class="label additionally__label"
        for="egg"
      >
        Яйца
      </label>
    </li>
  </ul>
  <button
    class="questionnaire__button button"
    type="submit"
  >
    отправить заявку
  </button>
  <p
    class="questionnaire__text"
  >
    * — Обязательные поля
  </p>
</div>
`;
